### DISCUSSION TITLE
Request bridge/devices after device joined

### DISCUSSION THREAD

**Discussion by asymetrixs** [Koenkk/zigbee2mqtt#22672] [General]
Hello,

perhaps I am mistaken but here is what I struggle with:
When I connect my MQTT client to Mosquitto and this in turn is connected to Zigbee2Mqtt, then after the connect took place, I receive the retained message for "bridge/devices".
Now I join a new Zigbee device to my network and at the end I get a message (bridge/event) that the interview was successful. This message does not match any message format used in "bridge/devices", moreover I cannot retrigger the publishing of "bridge/devices", so now I am kind of in a dirty state.
What I would like to have:
Either the message format of "bridge/event" when the interview was successful matches the one used in "bridge/devices", so that I can merge the data into the existing list OR I would like to have a way to make Zigbee2Mqtt reissue the "bridge/devices" list to get a full list of all connected things (yes, I could disconnect and reconnect but that is not pretty).
Eventually, after joining a new thing or removing an existing thing I'd like to somehow get an updated device list.

Is that possible, if so how?
How do you deal with it?

Thanks

**Comment by Koenkk** [comment]
Whenever a new device joins, a new message is published to`bridge/devices`
